packages/builders/src/components/v2/Container.ts

```diff
@@ -0,0 +1,184 @@
+
+	/**
+	 * Adds action row components to this container.
+	 *
+	 * @param input - The action row to add
+	 */
+	public addActionRowComponents(
+		...input: RestOrArray<
+			| ActionRowBuilder
+			| APIActionRowComponent<APIComponentInMessageActionRow>
+			| ((builder: ActionRowBuilder) => ActionRowBuilder)
+		>
+	): this {
+		const normalized = normalizeArray(input);
+		const resolved = normalized.map((component) => resolveBuilder(component, ActionRowBuilder));
+
+		this.data.components.push(...resolved);
+		return this;
+	}
+
+	/**
+	 * Adds file components to this container.
+	 *
+	 * @param input - The file components to add
+	 */
+	public addFileComponents(
+		...input: RestOrArray<APIFileComponent | FileBuilder | ((builder: FileBuilder) => FileBuilder)>
+	): this {
+		const normalized = normalizeArray(input);
+		const resolved = normalized.map((component) => resolveBuilder(component, FileBuilder));
+
+		this.data.components.push(...resolved);
+		return this;
+	}
+
+	/**
+	 * Adds media gallery components to this container.
+	 *
+	 * @param input - The media gallery components to add
+	 */
+	public addMediaGalleryComponents(
+		...input: RestOrArray<MediaGalleryBuilder | ((builder: MediaGalleryBuilder) => MediaGalleryBuilder)>
+	): this {
+		const normalized = normalizeArray(input);
+		const resolved = normalized.map((component) => resolveBuilder(component, MediaGalleryBuilder));
+
+		this.data.components.push(...resolved);
+		return this;
+	}
+
+	/**
+	 * Adds section components to this container.
+	 *
+	 * @param input - The section components to add
+	 */
+	public addSectionComponents(
+		...input: RestOrArray<SectionBuilder | ((builder: SectionBuilder) => SectionBuilder)>
+	): this {
+		const normalized = normalizeArray(input);
+		const resolved = normalized.map((component) => resolveBuilder(component, SectionBuilder));
+
+		this.data.components.push(...resolved);
+		return this;
+	}
+
+	/**
+	 * Adds separator components to this container.
+	 *
+	 * @param input - The separator components to add
+	 */
+	public addSeparatorComponents(
+		...input: RestOrArray<APISeparatorComponent | SeparatorBuilder | ((builder: SeparatorBuilder) => SeparatorBuilder)>
+	): this {
+		const normalized = normalizeArray(input);
+		const resolved = normalized.map((component) => resolveBuilder(component, SeparatorBuilder));
+
+		this.data.components.push(...resolved);
+		return this;
+	}
+
+	/**
+	 * Adds text display components to this container.
+	 *
+	 * @param input - The text display components to add
+	 */
+	public addTextDisplayComponents(
+		...input: RestOrArray<
+			APITextDisplayComponent | TextDisplayBuilder | ((builder: TextDisplayBuilder) => TextDisplayBuilder)
+		>
+	): this {
+		const normalized = normalizeArray(input);
+		const resolved = normalized.map((component) => resolveBuilder(component, TextDisplayBuilder));
+
+		this.data.components.push(...resolved);
+		return this;
+	}
+
```

Comment:
```suggestion

	/**
	 * Removes, replaces, or inserts components for this container
	 *
	 * @remarks
	 * This method behaves similarly
	 * to {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice | Array.prototype.splice()}.
	 *
	 * It's useful for modifying and adjusting order of the already-existing components of a container.
	 * @example
	 * Remove the first component:
	 * ```ts
	 * container.spliceComponents(0, 1);
	 * ```
	 * @example
	 * Remove the first n components:
	 * ```ts
	 * const n = 4;
	 * container.spliceComponents(0, n);
	 * ```
	 * @example
	 * Remove the last component:
	 * ```ts
	 * container.spliceComponents(-1, 1);
	 * ```
	 * @param index - The index to start at
	 * @param deleteCount - The number of components to remove
	 * @param components - The replacing component objects
	 */
	public spliceComponents(index: number, deleteCount: number, ...components: AnyActionRowComponentBuilder[]): this {
		this.data.components.splice(index, deleteCount, ...components);
		return this;
	}

	/**
	 * {@inheritDoc ComponentBuilder.toJSON}
	 */
```